pimd/pim_upstream.c

```diff
@@ -462,10 +462,26 @@ void pim_upstream_join_timer_decrease_to_t_override(const char *debug_label,
 		return;
 	}
 
-	join_timer_remain_msec = pim_time_timer_remain_msec(up->t_join_timer);
 	t_override_msec =
 		pim_if_t_override_msec(up->rpf.source_nexthop.interface);
 
+	if (up->t_join_timer) {
+		join_timer_remain_msec =
+			pim_time_timer_remain_msec(up->t_join_timer);
+	} else {
+		/* upstream join tracked with neighbor jp timer */
+		struct pim_neighbor *nbr;
+
+		nbr = pim_neighbor_find(up->rpf.source_nexthop.interface,
+					up->rpf.rpf_addr.u.prefix4);
+		if (nbr)
+			join_timer_remain_msec =
+				pim_time_timer_remain_msec(nbr->jp_timer);
+		else
+			/* Manipulate such that override takes place */
+			join_timer_remain_msec = t_override_msec + 1;
```

Comment:
i am a bit confused by the calculation of the override interval. according to the rfc it is supposed to be a random number between 0 and 2500 ms. what was the need for rounding up? is it because of the rounding down done by pim_if_t_override_msec?

Please add some comments here or in pim_if_t_override_msec to clarify for future.

Reply:
If join timer remaining is lesser than override time, we dont override.

If there is a case where neighbor is not found, remaining join time cannot be calculated. There is no other way to check remaining join time if join timer is not running. So the only option we have for sending join is to start the join timer and hope for neighbor to comeup before it expires.

This particular line is just to keep(manipulate) join timer more than override time so that the condition below that opts to start the join timer. 

There is no override time calculation change/round up here.